src/simplification/Simplifier.ts

```diff
@@ -16,52 +16,27 @@ export class Simplifier {
   options: SimplificationOptions;
   anonymCounter = 1;
   seenSchemas: Map<Schema, CommonModel> = new Map();
-
+  iteratedModels: {[key: string]: CommonModel} = {};
```

Comment:
`iteratedModels: Record<string, CommonModel> = {};`

Reply:
also, you should remove it and return this data from `ensureModelsAreSplit` function.

Reply:
I cant, since in the end I want to return all the models even from the recursive calls. Thats why this map is here to ensure that all separate models are kept and returned